.github/workflows/ci.yml

```diff
@@ -109,17 +101,13 @@ jobs:
           toolchain: stable
       - name: Chown cargo registry because of caching
         run: sudo chown -R runner ~/.cargo/registry
-      - name: Cache cargo registry
-        uses: actions/cache@v1
+      - name: Cache cargo registry and target
+        uses: actions/cache@v2
         with:
-          path: ~/.cargo/registry
-          key: registry-${{ hashFiles('**/Cargo.lock') }}
-      - name: Cache target
-        id: cache-target
-        uses: actions/cache@v1
-        with:
-          path: target
-          key: build-target-${{ steps.toolchain.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock') }}
+            path: |
+              ~/.cargo/registry
+              target
+            key: ${{ steps.toolchain.outputs.rustc_hash }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
It seems the cache is not scoped by jobs (it would be nice to check this just in case). 
And different cargo commands can possibly produce different binary output.
So I think it is safer to use different cache keys for different jobs.

Reply:
To be safe, I manually wrote out the job name in the cache key, so should be different for each now.